### DISCUSSION TITLE
Re-mapping bipartite graph connections after simplifying one of the sub graphs.

### DISCUSSION THREAD

**Discussion by dhorka** [pyg-team/pytorch_geometric#4529] [Q&A] [answered]
Hi,

I would like to ask you if you could help me to improve a problem that I have. Let me first setup the experiment. 
I have a bigraph with their relations defined in `edge_index_bigraph = [indx_g1, indx_g2]` where g1 is the first isolated graph and g2 is the second isolated graph. In this specific case, g1 is a point cloud. What I did is to sample the point cloud `g1` to a simplified version of it, lets called this simplified version `g1_simpl`. To relate the `g1` and `g1_simpl` I used the knn method of py_geo with `k=1`. Then what I get is a indx tensor that relates `g1`and `g1_simpl` let's called `g12g1simpl`

Now what I would like to do is create a new `edge_index_bigraph = [indx_g1_simpl, indx_g2]` which is based on the original connections of `edge_index_bigraph = [indx_g1, indx_g2]`. Basically what I need to do is to map and filter the old indexes with the new ones using the tensor `g12g1simpl`. Right now, the only way that I found to do that is using a for loop, however, this is quite slow. Do you know a better solution to solve that mapping problem that could be more efficient than using a classical loop approach?

Thanks!!

**Comment by rusty1s** [comment, marked as answer]
You should be able to do this via a simple `mapping` vector and some fancy indexing/masking logic:
```python
mapping = torch.full((indx_g1.size(0), ), -1, dtype=torch.long)
mapping[g1_simpl] = torch.arange(g1_simple.size(0))

indx_g1 = mapping[indx_g1]
mask = indx_g1 != -1
indx_g1 = indx_g1[mask]
```
